agent/provider/src/cli/rule.rs

```diff
@@ -95,14 +90,38 @@ fn set(set_rule: SetRule, config: ProviderConfig) -> Result<()> {
             SetOutboundRule::Disable => rules.set_enabled(false),
             SetOutboundRule::Enable => rules.set_enabled(true),
             SetOutboundRule::Everyone { mode } => rules.set_everyone_mode(mode),
-            SetOutboundRule::AuditedPayload { cert_id, mode } => match cert_id {
-                Some(_) => todo!("Setting rule for specific certificate isn't implemented yet"),
-                None => rules.set_default_audited_payload_mode(mode),
-            },
-            SetOutboundRule::Partner(PartnerRuleWithCert::CertId(CertId { cert_id, mode })) => {
+            SetOutboundRule::AuditedPayload(RuleWithCert::CertId(CertId { cert_id, mode })) => {
+                rules.set_audited_mode(cert_id, mode)
+            }
+            SetOutboundRule::AuditedPayload(RuleWithCert::ImportCert { import_cert, mode }) => {
+                let mut keystore = CompositeKeystore::load(&rules.cert_dir)?;
```

Comment:
This is unneeded as Keystore is inside rulestore already

Reply:
Right, I was confused because the `Partner` match-arm also created a new keystore. Corrected both.